src/graphics/webgl/webgl-shader.js

```diff
@@ -14,15 +14,7 @@ import { SHADERTAG_MATERIAL, semanticToLocation } from '../constants.js';
  */
 class WebglShader {
     constructor(shader) {
-
-        this.uniforms = [];
-        this.samplers = [];
-        this.attributes = [];
-
-        this.glProgram = null;
-        this.glVertexShader = null;
-        this.glFragmentShader = null;
-
+        this.loseContext();
```

Comment:
This looks weird to me. I'd rather than `loseContext` function was called something like `init` and `loseContext` called it.

Reply:
full agreed.